HeterogeneousCore/SonicCore/interface/SonicClientTypes.h

```diff
@@ -0,0 +1,27 @@
+#ifndef HeterogeneousCore_SonicCore_SonicClientTypes
+#define HeterogeneousCore_SonicCore_SonicClientTypes
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+//this base class exists to limit the impact of dependent scope in derived classes
+template <typename InputT, typename OutputT = InputT>
+class SonicClientTypes {
+public:
+  //typedefs for outside accessibility
+  typedef InputT Input;
+  typedef OutputT Output;
+  //destructor
+  virtual ~SonicClientTypes() {}
+
+  //accessors
+  Input& input() { return input_; }
+  const Input& input() const { return input_; }
```

Comment:
this is not used, AFAICT; is there a use case for inspecting the input value in a `const` context ?